extension/src/background/messageListener/popupMessageListener.ts

```diff
@@ -294,6 +295,26 @@ export const popupMessageListener = (
     });
   };
 
+  /* @TODO add toggle for Mainnet & Testnet */
+  const saveSettings = () => {
+    const { isDataSharingAllowed } = request;
+
+    localStorage.setItem(DATA_SHARING_ID, JSON.stringify(isDataSharingAllowed));
+
+    sendResponse({
+      isDataSharingAllowed,
+    });
+  };
+
+  const loadSettings = () => {
+    const dataSharingValue = localStorage.getItem(DATA_SHARING_ID) || "true";
```

Comment:
defaulting to "true". a user can opt out after account creation has been completed, per ticket